internal/internal_event_handlers.go

```diff
@@ -172,19 +172,21 @@ type (
 
 	localActivityTask struct {
 		sync.Mutex
-		workflowTask    *workflowTask
-		activityID      string
-		params          *ExecuteLocalActivityParams
-		callback        LocalActivityResultHandler
-		wc              *workflowExecutionContextImpl
-		canceled        bool
-		cancelFunc      func()
-		attempt         int32  // attempt starting from 1
-		attemptsThisWFT uint32 // Number of attempts started during this workflow task
-		pastFirstWFT    bool   // Set true once this LA has lived for more than one workflow task
-		retryPolicy     *RetryPolicy
-		expireTime      time.Time
-		header          *commonpb.Header
+		workflowTask                *workflowTask
+		activityID                  string
+		params                      *ExecuteLocalActivityParams
+		callback                    LocalActivityResultHandler
+		wc                          *workflowExecutionContextImpl
+		canceled                    bool
+		cancelFunc                  func()
+		attempt                     int32  // attempt starting from 1
+		attemptsThisWFT             uint32 // Number of attempts started during this workflow task
+		pastFirstWFT                bool   // Set true once this LA has lived for more than one workflow task
+		retryPolicy                 *RetryPolicy
+		expireTime                  time.Time
+		scheduledTime               time.Time // Time the activity was scheduled initially.
+		currentAttemptScheduledTime time.Time // Time this attempt of the activity was scheduled.
```

Comment:
This field only ever looks set, never read. Am I missing some code or context?

Reply:
yeah good catch i'll remove. I had it in to implement `ScheduleToClose` timeout, but left that out of this  PR